src/app.py

```diff
@@ -100,24 +100,28 @@ def appium_run(avd_name: str):
     :param avd_name: Name of android virtual device / emulator
     """
     cmd = 'appium'
+    local_ip = os.popen('ifconfig eth0 | grep \'inet addr:\' | cut -d: -f2 | awk \'{ print $1}\'').read().strip()
 
     grid_connect = convert_str_to_bool(str(os.getenv('CONNECT_TO_GRID', False)))
     logger.info('Connect to selenium grid? {connect}'.format(connect=grid_connect))
     if grid_connect:
         try:
-            appium_host = os.getenv('APPIUM_HOST', '127.0.0.1')
+            mobile_web_test = convert_str_to_bool(str(os.getenv('MOBILE_WEB_TEST', False)))
+            default_web_browser = os.getenv('BROWSER')
+            appium_host = os.getenv('APPIUM_HOST', local_ip)
             appium_port = int(os.getenv('APPIUM_PORT', 4723))
             selenium_host = os.getenv('SELENIUM_HOST', '172.17.0.1')
             selenium_port = int(os.getenv('SELENIUM_PORT', 4444))
-            create_node_config(avd_name, appium_host, appium_port, selenium_host, selenium_port)
+            browser_name = default_web_browser if mobile_web_test else 'android'
```

Comment:
The browser value can actually be 3 things - chrome, browser or android.
`chrome` is default web browser for android 7.0+.
`browser` is default android browser for 6.0 and older.
`android` is when MOBILE_WEB_TEST=false, essentially it is for setup when web browser capability is not needed and it also allows for selenium hub to display android icon in hubs console.
Potentially it is possible to ditch the `browser` for older android versions and just run web tests with capabilties defined like:
```
'appPackage': 'com.android.browser',
'appActivity': 'com.android.browser.BrowserActivity',
```
but at least with appium java client I have encountered some issues when using browser app this way, thus I decided that populating browserName capability with either `chrome` or `browser` is better. Hope this makes it a bit more clear :D